cinema/views.py

```diff
@@ -1 +1,61 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import CinemaHall, Genre, Actor, Movie, MovieSession
+from cinema.serializers import (
+    CinemaHallSerializer, GenreSerializer,
+    ActorSerializer, MovieSerializer,
+    MovieSessionSerializer, MovieListSerializer,
+    MovieDetailSerializer, MovieSessionListSerializer,
+    MovieSessionDetailSerializer
+)
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            return queryset.prefetch_related("genres", "actors")
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.all()
+    serializer_class = MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action == "list":
+            return queryset.select_related("movie", "cinema_hall")
```

Comment:
shouldn't we use select_related when action is retrieve and we show movie and hall for a specific moviesession?

Reply:
I got